### PR TITLE
Standardize use of Paginator limits

### PR DESCRIPTION
**Pull request by EricBorczuk**
The "search within a document" functionality now properly honors the page-size default of 1. That default could change in the future, but is now shared with the "search across collections" functionality.

Note that In order to do this, the db page size actually had to be set to be equal to the doc page size when performing search within a document.  This is so that no rows are discarded in memory after querying and the pagination state is maintained properly.

### CODE REVIEW

File: restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java
Comment (dimas-b) [inline restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java:663]: What about ` defaultValue = "100"` for the `page-size` HTTP parameter on line 587? Why is it not effective? Is it just for swagger documention? In the latter case I think we ought to fix it too :)

Comment (EricBorczuk) [reply restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java:663]: Oh yes - my fault - that is just for swagger, will change

**Review by dimas-b — APPROVED** [review]

File: grpc/pom.xml
Comment (dimas-b) [inline grpc/pom.xml:7]: This should help, but #926 takes priority for updating this version on `master`, I guess

Comment (EricBorczuk) [reply grpc/pom.xml:7]: Yea I will wait until after #926 to merge

Comment (dimas-b) [reply grpc/pom.xml:7]: #926 has been merged

File: restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java
Comment (dimas-b) [inline restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java:1222]: Is this no longer an issue? It is taking me some extra efforts to implement on my branch :smiley: So, I'll wait for this to merge and rebase my refactoring.

**Review by dimas-b — APPROVED** [review]

**Comment by dimas-b** [comment]
@EricBorczuk : if test are causing trouble, we can probably just pull the default page size fixes into my branch and I will open a PR from it. I'm almost ready and I have all Docs inntegratio tests passing except `testPaginationDisallowedLimitedSupport` but you removed it anyway. Let's discuss tomorrow.

**Comment by EricBorczuk** [comment]
Sounds good - yea, I just realized that searching within a document shouldn't have the same max limit as searching across collections (especially because of how limited support filters work when searching within a document) - the limit of 20 isn't enough in some cases, and I only made a limit for the collection search because documents could be arbitrarily large.

But yeah, we will discuss tomorrow :) I suspect this test run will succeed though

File: restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java
Comment (EricBorczuk) [reply restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java:1222]: So - I realized along the way that this can be left out - the page size for searching within a document just works in a strange way for limited support filters as it is...but I think it is fine to remove this behavior, it is too restrictive as it is

File: restapi/src/main/java/io/stargate/web/docsapi/dao/Paginator.java
Comment (tomekl007) [inline restapi/src/main/java/io/stargate/web/docsapi/dao/Paginator.java:33]: should we keep the upper `pageSize` validation as well (besides the lower bound)?

File: testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java
Comment (tomekl007) [inline testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java:1375]: can we add two test cases for edge conditions:
1. Validate that it returns an error when page-size > upper bound
2. Validate that it returns an error when page-size < lower bound

File: restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java
Comment (dougwettlaufer) [inline restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java:762]: Looks like Paginator already does this? 
https://github.com/stargate/stargate/blob/master/restapi/src/main/java/io/stargate/web/docsapi/dao/Paginator.java#L32

File: restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java
Comment (dougwettlaufer) [inline restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java:770]: Also duplicated by Paginator constructor?

File: restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java
Comment (dougwettlaufer) [inline restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java:665]: Why does this one get `pageSize` twice? Just curious.

Would it make sense to have another constructor that takes 3 args?

**Comment by ivansenic** [comment]
> #937 implements the same page size defaults. As discussed, @EricBorczuk , would you be ok with closing this PR in favour of #37 and perhaps rebased it after #937 merges and re-apply whatever is not covered by #937? WDYT?

@EricBorczuk Also, I just started on adding validation layer to the resources, maybe you can completely skip this and let me handle it there..

**Pull request closed without merging**
